test/node/src/transaction.test.ts

```diff
@@ -154,6 +156,24 @@ for (const dialect of DIALECTS) {
       }
     })
 
+    it('should retain the full stack trace', async () => {
+      try {
+        await ctx.db.transaction().execute(async (trx) => {
+          await trx.selectFrom('person').where('id', 'in', -1).execute()
+        })
+
+        expect.fail('Expected transaction to fail')
+      } catch (error) {
+        if (dialect === 'sqlite') {
+          expect(error).to.be.instanceOf(SqliteError)
+        } else if (dialect === 'postgres') {
+          expect(error).to.be.instanceOf(PostgresError)
+        }
```

Comment:
not very familiar with chai; maybe there's a cleaner way to assert this

Reply:
More just to make sure the error being thrown is a database error and not an error in our test code (in which case the assertion below would still pass).